bundles/framework/statehandler/instance.js

```diff
@@ -495,6 +497,30 @@ Oskari.clazz.define('Oskari.mapframework.bundle.statehandler.StateHandlerBundleI
             for (i = 0; i < selectedLayers.length; i += 1) {
                 sandbox.request(module.getName(), rbRemove(selectedLayers[i].getId()));
             }
+        },
+        _setupActionsOnUserActivity (sessionLengthInMinutes) {
+            const throttleTime = 300000;
+            const resetRemainingSessionTime = Oskari.util.throttle(() => {
+                jQuery.ajax({
+                    type: 'GET',
+                    url: Oskari.urls.getRoute('ResetRemainingSessionTime'),
+                    error: (jqXHR, textStatus, errorThrown) => {
+                        const errorText = Oskari.util.getErrorTextFromAjaxFailureObjects(jqXHR, errorThrown);
+                        this._log.error(errorText);
+                    },
+                    success: (response) => {
+                        this._log.debug(response);
+
+                        if (Oskari.user().isLoggedIn() && sessionLengthInMinutes > 0) {
```

Comment:
This if statement could be moved to the beginning of this method and if the user is not logged in -> skip the rest of the method.

Reply:
Good point. Method updated as recommended.